docs/source/l1_to_l2b_e2e.rst

```diff
@@ -0,0 +1,198 @@
+
+      # Or specify custom paths
+      python tests/test_l1_to_l2b.py --tvacdata_dir /path/to/CGI_TVAC_Data --outputdir /path/to/output
+
+
+Test Setup
+~~~~~~~~~~
+
+To run this test, you will need three specific datasets from the database:
+
+1. **L1 Data** - Raw Level 1 data files.
+2. **L2b Data** - Processed Level 2b data files.
+3. **Cals** - Calibration files required for processing.
+
+If you don't have all the datasets downloaded, make sure you at least have these three files to run the test promptly. The script imports necessary libraries and determines the directory containing the script.
+   
+.. code-block:: python
+
+    import argparse
+    import os
+    import pytest
+    import numpy as np
+    import astropy.time as time
+    import astropy.io.fits as fits
+    import corgidrp
+    import corgidrp.data as data
+    import corgidrp.mocks as mocks
+    import corgidrp.walker as walker
+    import corgidrp.caldb as caldb
+    import corgidrp.detector as detector
+
+    thisfile_dir = os.path.dirname(__file__)  # this file's folder
+
+The `test_l1_to_l2b` function initializes paths for input data, calibration files, and output directories.
+
+.. code-block:: python
+
+    @pytest.mark.e2e
+    def test_l1_to_l2b(tvacdata_path, e2eoutput_path):
+        # Define input data paths
+        l1_datadir = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "L1")
+        l2b_datadir = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "L2b")
+        processed_cal_path = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "Cals")
+
+        # Create output directory if it does not exist
+        l2b_outputdir = os.path.join(e2eoutput_path, "l1_to_l2b_output")
+        if not os.path.exists(l2b_outputdir):
+            os.mkdir(l2b_outputdir)
+
+Calibration files such as non-linearity tables, dark current, flat fields, and bad pixel maps are loaded.
+
+.. code-block:: python
+
+        nonlin_path = os.path.join(processed_cal_path, "nonlin_table_240322.txt")
+        dark_path = os.path.join(processed_cal_path, "dark_current_20240322.fits")
+        flat_path = os.path.join(processed_cal_path, "flat.fits")
+        fpn_path = os.path.join(processed_cal_path, "fpn_20240322.fits")
+        cic_path = os.path.join(processed_cal_path, "cic_20240322.fits")
+        bp_path = os.path.join(processed_cal_path, "bad_pix.fits")
+
+The raw science data files and mock calibration files are defined.
+
+.. code-block:: python
+
+        l1_data_filelist = [os.path.join(l1_datadir, "{0}.fits".format(i)) for i in [90499, 90500]]
+        mock_cal_filelist = [os.path.join(l1_datadir, "{0}.fits".format(i)) for i in [90526, 90527]]
+        tvac_l2b_filelist = [os.path.join(l2b_datadir, "{0}.fits".format(i)) for i in [90529, 90531]]
+
+The calibration database (`caldb`) is initialized, and calibration entries are created.
+
+.. code-block:: python
+
+        pri_hdr, ext_hdr = mocks.create_default_headers()
```

Comment:
this is all mocking code for squeezeing the test data into the official format, so it should be noted as such.

Reply:
I think this could be better clarified in the text.